src/main/java/com/hierynomus/msfscc/fileinformation/FileInformationFactory.java

```diff
@@ -37,7 +38,7 @@
         renBuf.putRawBytes(new byte[]{0, 0, 0, 0, 0, 0, 0});
         renBuf.putUInt64(0);
         renBuf.putUInt32(newName.length() * 2); // unicode
-        renBuf.putRawBytes(newName.getBytes(StandardCharsets.UTF_16));
+        renBuf.putRawBytes(SMB2Functions.unicode(newName));
```

Comment:
By depending on the `SMB2Functions` you introduce a circular package dependency. Prefer just to use the right `StandardCharsets` constant.